### PR TITLE
chore: standardize npm script names

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Refers https://github.com/eslint/eslint/issues/14827

#### What changes did you make? (Give an overview)

This updates the names of the scripts in package.json to be consistent with the [new standard](https://eslint.org/docs/latest/contribute/package-json-conventions).

#### Is there anything you'd like reviewers to focus on?
We may need to update the `release` related scripts on Jenkins before merging this PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,21 +46,21 @@
     "typescript": "^4.6.2"
   },
   "scripts": {
-    "prepare": "npm run build",
-    "bundle": "rollup -c",
     "build": "npm run bundle && npm run tsc",
     "build:debug": "npm run bundle -- -m && npm run tsc",
+    "build:keys": "node tools/build-keys-from-ts",
+    "bundle": "rollup -c",
```
Comment (mdjermanovic) [inline package.json:52]: This could be `build:cjs`

File: package.json [hunk 2]
```diff
@@ -46,21 +46,21 @@
...
+    "release:publish": "eslint-publish-release",
     "test": "mocha tests/lib/**/*.cjs && c8 mocha tests/lib/**/*.js && npm run tsd",
-    "coverage": "c8 report --reporter lcov && opener coverage/lcov-report/index.html",
-    "generate-release": "eslint-generate-release",
-    "generate-alpharelease": "eslint-generate-prerelease alpha",
-    "generate-betarelease": "eslint-generate-prerelease beta",
-    "generate-rcrelease": "eslint-generate-prerelease rc",
-    "publish-release": "eslint-publish-release"
+    "tsc": "tsc",
+    "tsd": "tsd"
```
Comment (mdjermanovic) [inline package.json:63]: Maybe `build:types` and `test:types`.

File: package.json [hunk 3]
```diff
@@ -46,21 +46,21 @@
     "typescript": "^4.6.2"
   },
   "scripts": {
-    "prepare": "npm run build",
-    "bundle": "rollup -c",
     "build": "npm run bundle && npm run tsc",
     "build:debug": "npm run bundle -- -m && npm run tsc",
+    "build:keys": "node tools/build-keys-from-ts",
+    "bundle": "rollup -c",
+    "coverage": "c8 report --reporter lcov && opener coverage/lcov-report/index.html",
```
Comment (mdjermanovic) [inline package.json:53]: Not sure about this one, maybe `test:open-coverage`?

**Review by mdjermanovic — COMMENTED** [review]
We should generally classify scripts as build, release, lint, start, or test.

File: package.json (hunk 3 above)
Comment (snitin315) [reply package.json:53]: Makes sense.

File: package.json (hunk 1 above)
Comment (snitin315) [reply package.json]: Updated.

File: package.json (hunk 2 above)
Comment (snitin315) [reply package.json]: Updated.

File: package.json [hunk 4]
```diff
@@ -46,21 +46,21 @@
...
-    "generate-rcrelease": "eslint-generate-prerelease rc",
-    "publish-release": "eslint-publish-release"
+    "prepare": "npm run build",
+    "release:generate:latest": "eslint-generate-release",
+    "release:generate:alpha": "eslint-generate-prerelease alpha",
+    "release:generate:beta": "eslint-generate-prerelease beta",
+    "release:generate:rc": "eslint-generate-prerelease rc",
+    "release:publish": "eslint-publish-release",
+    "test": "mocha tests/lib/**/*.cjs && c8 mocha tests/lib/**/*.js && npm run test:types",
+    "test:open-coverage": "c8 report --reporter lcov && opener coverage/lcov-report/index.html",
```
[the remark this answers is not in the document]
Comment (snitin315) [reply package.json:62]: Updated 👍🏻

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Would like @nzakas to verify before merging. I can update Jenkins script after we merge this.

**Comment by mdjermanovic** [comment]
> We may need to update the `release` related scripts on Jenkins before merging this PR.

Updated.

**Pull request merged**